stdlib/Statistics/test/runtests.jl

```diff
@@ -86,6 +86,10 @@ end
     @test ismissing(mean([missing, NaN]))
     @test isequal(mean([missing 1.0; 2.0 3.0], dims=1), [missing 2.0])
     @test mean(skipmissing([1, missing, 2])) === 1.5
+    @test isequal(mean(Complex{Float64}[]), NaN+NaN*im)
```

Comment:
This PR is in muddy waters :) You cannot use `===` as we have different `NaN` in Julia and here it would fail. Notice that:
```
julia> NaN === -NaN
false
```
and this is exactly the case here, that is why I had to use `isequal`.

I will add the tests.

Reply:
I must be missing something: you really expect `NaN::Float64` here, right?

Reply:
The situation is more complex (and that is why I have used this as a test):
```
julia> x = mean(Complex{Float64}[])
NaN + NaN*im

julia> bitstring(x.re)
"1111111111111000000000000000000000000000000000000000000000000000"

julia> bitstring(x.im)
"1111111111111000000000000000000000000000000000000000000000000000"

julia> bitstring(NaN) # see that the bitstring is different
"0111111111111000000000000000000000000000000000000000000000000000"
```

also you have:
```
julia> x = mean(Int[])
NaN

julia> x === NaN
false
```
For the same reason that we have different `NaN`s in Julia (and as you see they actually happen in practice).